### PR TITLE
RUN: Implicitly pass required features to a run configuration

### PR DESCRIPTION
**Pull request by mchernyavsky**
Closes https://github.com/intellij-rust/intellij-rust/issues/6262.

### CODE REVIEW

File: src/test/resources/org/rust/cargo/runconfig/producers/fixtures/bench_configuration_uses_default_environment.xml [hunk 1]
```diff
@@ -2,6 +2,7 @@
   <configuration name="Bench bench_foo" type="CargoCommandRunConfiguration" factoryName="Cargo Command">
     <option name="channel" value="NIGHTLY" />
     <option name="command" value="bench --package test-package --lib bench_foo -- --exact" />
+    <option name="requitedFeatures" value="false" />
```
Comment (Kobzol) [inline src/test/resources/org/rust/cargo/runconfig/producers/fixtures/bench_configuration_uses_default_environment.xml:5]: Typo in `required` (also in other XMLs).

File: src/main/kotlin/org/rust/cargo/runconfig/command/CargoCommandConfiguration.kt [hunk 2]
```diff
@@ -65,6 +67,7 @@ class CargoCommandConfiguration(
         super.writeExternal(element)
         element.writeEnum("channel", channel)
         element.writeString("command", command)
+        element.writeBool("requitedFeatures", requiredFeatures)
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/cargo/runconfig/command/CargoCommandConfiguration.kt:70]: Typo in "requitedFeatures"

File: src/main/kotlin/org/rust/cargo/runconfig/command/CargoCommandConfiguration.kt [hunk 3]
```diff
@@ -212,6 +218,52 @@ class CargoCommandConfiguration(
         fun findCargoProject(project: Project, cmd: String, workingDirectory: Path?): CargoProject? = findCargoProject(
             project, ParametersListUtil.parse(cmd), workingDirectory
         )
+
+        fun findCargoPackage(
+            cargoProject: CargoProject,
+            additionalArgs: List<String>,
+            workingDirectory: Path?
+        ): CargoWorkspace.Package? {
+            val packages = cargoProject.workspace?.packages.orEmpty()
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/cargo/runconfig/command/CargoCommandConfiguration.kt:227]: Do I understand correctly that it's impossible to run Cargo with a package outside of a workspace? Hence, we could filter workspace-only packages here.

**Review by vlad20012 — APPROVED** [review]

**Review by mchernyavsky — COMMENTED** [review]
bors r=vlad20012

**Pull request merged**